### PR TITLE
feat(discover2): Move buttons to add/edit columns out of the Grid

### PR DESCRIPTION
**Pull request by leedongwei**
- This commit is part of a series of small changes that'll bring the table UIUX closer to its intended design
- This commit is dependent on dc058b0 or #15682 to be merged

### Before
image

### After
image
image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx [hunk 1]
```diff
@@ -141,86 +141,77 @@ class GridEditable<
...
+    }
+
+    const sorryMessage = t('This will be improved soon! Please bear with us!');
+
+    if (!this.state.isEditing) {
+      return (
+        <ToolTip title={sorryMessage}>
+          <HeaderButton onClick={this.toggleEdit}>
+            <InlineSvg src="icon-edit-pencil" />
+            Edit Columns
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/index.tsx]:
```suggestion
            {t('Edit Columns')}
```

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx [hunk 2]
```diff
@@ -141,86 +141,77 @@ class GridEditable<
...
-            <EmptyStateWarning>
-              <p>{t('No results found')}</p>
-            </EmptyStateWarning>
-          </GridBodyCellSpan>
-        </GridRow>
-      </GridBody>
+      <ToolTip title={sorryMessage}>
+        <HeaderButton onClick={this.toggleEdit}>
+          <InlineSvg src="icon-circle-close" />
+          Exit Edit
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/index.tsx]:
```suggestion
          {t('Exit Edit')}
```

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx [hunk 3]
```diff
@@ -141,86 +141,77 @@ class GridEditable<
...
-        </GridRow>
-      </GridBody>
-    );
-  };
+  renderGridHeadEditButtons = () => {
+    if (!this.props.isEditable) {
+      return null;
+    }
+
+    const sorryMessage = t('This will be improved soon! Please bear with us!');
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/index.tsx]: Are you planning to do follow up pull requests to remove the need to apologize?

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx [hunk 4]
```diff
@@ -333,24 +280,82 @@ class GridEditable<
...
+    const {title, isEditable} = this.props;
 
     return (
-      <GridPanel>
-        <GridPanelBody>
+      <React.Fragment>
+        <Header>
+          {/* TODO(leedongwei): Check with Bowen/Dora on what they want the
+          default title to be */}
+          <HeaderTitle>{title || 'Query Builder'}</HeaderTitle>
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/index.tsx]:
```suggestion
          <HeaderTitle>{title || t('Query Builder')}</HeaderTitle>
```

File: src/sentry/static/sentry/app/components/gridEditable/styles.tsx [hunk 5]
```diff
@@ -28,10 +25,43 @@ type GridEditableProps = {
   isDragging?: boolean;
 };
 
-export const GridPanel = styled(Panel)`
-  /* overflow: hidden; */
+export const Header = styled('div')`
+  display: flex;
+  /* flex-direction: ; */
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/styles.tsx]: Could just delete this.

**Comment by dashed** [comment]
@leedongwei Hey Danny, I'm wondering if you're still pushing updates to this pull-request/branch.

**Comment by leedongwei** [comment]
@dashed Yea, I'm coming back to this.

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx (hunk 3 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx]: There was a chance that the table might be in a limbo state for a while but it's not necessary now.

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx [hunk 6]
```diff
@@ -333,24 +274,82 @@ class GridEditable<
     );
   };
 
+  renderError = () => {
+    const {error} = this.props;
+
+    return (
+      <GridBody>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/gridEditable/index.tsx]: These seem to be repeated a few times, is it possible to move them up to the call site?

File: src/sentry/static/sentry/app/components/gridEditable/styles.tsx [hunk 7]
```diff
@@ -28,10 +25,42 @@ type GridEditableProps = {
...
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  margin-bottom: ${space(1)};
+`;
+export const HeaderTitle = styled('h2')`
+  margin: 0;
+
+  font-size: 20px;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/gridEditable/styles.tsx]: Is there a size in `theme`?

File: src/sentry/static/sentry/app/components/gridEditable/styles.tsx [hunk 8]
```diff
@@ -28,10 +25,42 @@ type GridEditableProps = {
...
+  font-weight: normal;
+  color: ${p => p.theme.gray3};
 `;
-export const GridPanelBody = styled(PanelBody)``;
+export const HeaderButton = styled('div')`
+  display: flex;
+  align-items: center;
+
+  color: ${p => p.theme.gray3};
+  cursor: pointer;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/gridEditable/styles.tsx]: Can/should this be a `<Button>`?

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Comment by dashed** [comment]
@leedongwei I think you may want to rebase against `master`.

File: src/sentry/static/sentry/app/components/gridEditable/styles.tsx (hunk 8 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/gridEditable/styles.tsx]: Might be borderless `Button` but this component will be changed in an upcoming PR.

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx (hunk 6 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx]: Good catch.

File: src/sentry/static/sentry/app/components/gridEditable/styles.tsx (hunk 7 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/gridEditable/styles.tsx]: It's `18px` and `22px` in the theme file but `20px` when guesstimating from the Figma design. I've changed to use `theme` and will let @doralchan tweak it.

**Comment by dashed** [comment]
@leedongwei this would need to be addressed:

```
ERROR in /home/travis/build/getsentry/sentry/src/sentry/static/sentry/app/components/gridEditable/index.tsx
[tsl] ERROR in /home/travis/build/getsentry/sentry/src/sentry/static/sentry/app/components/gridEditable/index.tsx(9,1)
      TS6133: 'ToolTip' is declared but its value is never read.
error Command failed with exit code 2.
```

**Pull request merged**